woocommerce/woocommerce
# Filter catalog products by attribute using the new lookup table

### All Submissions:


### Changes proposed in this Pull Request:

Implements the filtering showcased in https://github.com/woocommerce/woocommerce/pull/29190 on the foundation laid by https://github.com/woocommerce/woocommerce/pull/29778.

Closes #29610.

### How to test the changes in this Pull Request:

1. Create a few products, both simple and variable, with some variations set out of stock. The sample product data included in [the PoC PR](https://github.com/woocommerce/woocommerce/pull/29190) may be useful (if you use that dataset, take a look at p7bje6-31P-p2 too).
2. Follow the testing instructions of https://github.com/woocommerce/woocommerce/pull/29778. Make sure to enable the lookup table usage after the table creation and filling finishes.
3. Add the appropriate "filter product by attribute" widgets and try various filtering combinations, with and without the "Hide out of stock products" setting.
4. Also test with the lookup table usage disabled, it should still work exactly as it does in `trunk`.

**Important note:** live sync between products and the lookup table isn't in place yet, thus if you do any change on existing products or variations (or create or delete products or variations) you'll need to fully regenerate the lookup table again.

### Other information:


### Changelog entry

> Add - Product attributes lookup table usage when enabled.
